prime_factor/prime_factor.c

```diff
@@ -0,0 +1,40 @@
+#include <stdio.h>
+#include <string.h> // for memset
+#include <stdbool.h> // for bool type
+
+int prime_factors(int num, int *ar) {
+    int cnt = 0; // index to the last location in the array
+    bool flag[num];
+    memset(flag, false, sizeof(flag));
```

Comment:
This suggests that flag will be false if number is a prime (factor). This can be tricky to understand at places. I think that it will be better if we say that true indicates a prime (factor).

What do you think @yashLadha?